src/net-to-gross/calcTaxableIncome.js

```diff
@@ -0,0 +1,114 @@
+const Big = require('big.js');
+const {TAX_RATES, DEDUCTION_PER_PERSON, SELF_DEDUCTION} = require('../salaryConstants');
+/**
+ * Gross = Net + totalTax + totalInsurance
+ * Unit: a million (Ex: 11 means 11 million)
+ *
+ * Taxable Income (ti) = net + totalTax - 11; Total tax is the sum of the tax amounts at each level (tax rate * deduction).
+ * ti = net + sum tax amounts (tax rate * deduction) - 11
+ *
+ * Each tax level has different rate and its max deduction. Therefore, we have below un-equations
+ *
+ * lv1 ti ∈ (0, 5]:
+ * ==> 0 < net + 5% * min(ti, 5) - 11 <= 5
+ * ==> 11 - 5% * min(0, 5) < net <= 5 + 11 - 5% * min(5, 5)
+ * ==> 11 < net <= 15.75
+ *
+ * lv2: ti <= net + 5% * 5 + 10% *min(ti - 5, 5) - 11
+ * lv3: ti = net + 5% * 5 + 10% * 5 + 15% *min(ti - 10, 8) - 11
+ * lv4: ti <= net + 5% * 5 + 10% * 5 + 15%*8 + 20% *min(ti-18, 14)- 11
+ * lv5: ti <= net + 5% * 5 + 10% * 5 + 15%*8 + 20%*14 + 25%*min(ti-32, 20) - 11
+ * lv6: ti <= net + 5% * 5 + 10% * 5 + 15%*8 + 20%*14 + 25%*20 +30%*min(ti-52, 28) - 11
+ * lv7: ti >= net + 5% * 5 + 10% * 5 + 15%*8 + 20%*14 + 25%*20 +30%*28 + 35%*min(ti-80) - 11
+ *
+ *
+ * @type {[{taxableIncomeFactor: number, inclusiveNetMax: number, totalDeductionTimeRatePrev: number}]}
+ */
+
+const TAX_FACTOR_RANGES = [
+    // lv1 ti ∈ (0, 5], rate 5%:
+    // ==> 0 < net + 5% * min(ti, 5) - 11 <= 5
+    // ==> 11 - 5% * min(0, 5) < net <= 5 + 11 - 5% * min(5, 5)
+    // ==> 11 < net <= 15.75
+
+    //==>  ti = (net - 11)/0.95 when net ∈ (11, 15.75]
+    {totalDeductionTimeRatePrev: 11_000_000, inclusiveNetMax: 15_750_000, taxableIncomeFactor: 0.95},
+
+    // lv2 ti ∈ (5, 10], rate 10%:
+    // ==> 5 < net + 5% * 5 + 10% *min(ti - 5, 5) - 11 <= 10
+    // ==> 5 + 11 - 5% * 5 - 10% *min(5 - 5, 5) < net <= 10 + 11 - 5 *5% - 10% *min(10 - 5, 5)
+    // ==> 15.75 < net <= 20.25
+
+    // ti = net + 5% * 5 + 10% *min(ti - 5, 5) - 11
+    // 0.9ti = net + 5% * 5 - 10% * 5 - 11
+    // ti = (net - 11.25)/0.9 when net ∈ (15.75, 20.25]
+    {totalDeductionTimeRatePrev: 11_250_000, inclusiveNetMax: 20_250_000, taxableIncomeFactor: 0.9},
+
+    /**
+     * Following above inequation, we can conclude for the rest of tax rates.
+     */
+
+    // lv3 ti ∈ (10, 18], rate 15%:
```

Comment:
shorten the formula for the rest of the tax rates. Think about the idea to include in `salaryConstants.js`